pkg/apis/core/types.go

```diff
@@ -3626,6 +3626,13 @@ type PodStatus struct {
 	// +featureGate=InPlacePodVerticalScaling
 	// +optional
 	Resize PodResizeStatus
+
+	// TotalResourcesRequested is the effective resource requests for this pod taking into consideration init
+	// and regular containers, and the pod overhead. If there is no effective resource request due to zero pod overhead
+	// and no requests on init or regular containers, the field is omitted.
+	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	// +optional
+	TotalResourcesRequested ResourceList
```

Comment:
I wonder if TotalRequestedResources will be more aligned with other naming like https://github.com/kubernetes/kubernetes/pull/116450

Reply:
Maybe so? I think @vinaykul suggested this name, any thoughts?

Reply:
We [recently changed the name](https://github.com/kubernetes/kubernetes/pull/116450) `ResourcesAllocated` to `AllocatedResources` [for consistency](https://kubernetes.slack.com/archives/CJUQN3E4T/p1678415227264449?thread_ts=1678401527.788819&cid=CJUQN3E4T). (Adjective-Noun). So +1 for TotalRequestedResources (sorry!)
/cc @thockin

Reply:
No problem, curious what you think about the semantics.  In the event of a resize that won't work, should it report the requested size or the in-use size?  There's some discussion below with Sergey, I can see utility for both.

Reply:
Imho it should be what we want (what the pod wants in order to successfully schedule taking into account init-container resources, pod overhead, regular container resources). `Requested` in the name makes that pretty clear.

Adding in-place pod resize to the mix, the signal from the summarized PodStatus.Resize is an easy signal to the user that informs how the `Requested` resize is going. `InProgress` = on the way, "" = done/no resize pending, `Proposed` = ACK we heard you,  `Deferred` = sure but not right now (evict and go elsewhere if you cant wait), `Infeasible` = not gonna happen, evict and go elsewhere to get what you need.